include/chemfiles/files/GzFile.hpp

```diff
@@ -34,6 +34,9 @@ class GzFile final: public TextFileImpl {
     gzFile file_ = nullptr;
 };
 
-}
+/// Inflates a GZipped file from a `src` buffer
+std::vector<char> gzinflate_in_place(char* src, size_t size);
```

Comment:
From my understanding, a `std::vector` will copy the contents of an array if it is given in the associated constructor. I want to avoid this copy and so wrap and these functions (which probably need better names) take the original `char*` argument which originates from the `memory_reader` function. If there's a better way, I am all for it.

Reply:
Oh, ok. It should maybe take a `const char*` though